cloudant-sync-datastore-core/src/test/java/com/cloudant/sync/internal/documentstore/AttachmentTest.java

```diff
@@ -411,4 +414,23 @@ public void attachmentOrderingTest() throws Exception {
         Matcher m = p.matcher(regex);
         Assert.assertTrue("Should match pattern", m.matches());
     }
+
+    // test for regression on fetching attachments after deleting revision
+    @Test
+    public void getAttachmentsOnDeletedRevision() throws Exception {
```

Comment:
Might be worth having a similar test that has some attachments before the deletion too, but I don't think it is essential as this already covers the `CCE` problem.